python/cudf/cudf/core/table.py

```diff
@@ -1,7 +1,38 @@
-from cudf._libxx.table import _Table
+import pandas as pd
 
+import cudf._libxx as libcudfxx
+from cudf.core.column import as_column, build_categorical_column
+from cudf.utils.dtypes import is_categorical_dtype
+
+
+class Frame(libcudfxx.Table):
+    def gather(self, gather_map):
+        if not pd.api.types.is_integer_dtype(gather_map.dtype):
```

Comment:
Should we typecheck gather_map here or is that being done somewhere else already?

Reply:
nvm being typechecked down in Table